### DISCUSSION TITLE
The easiest way to do a safely remove hardware and eject media

### DISCUSSION THREAD

**Discussion by abakosh** [hathach/tinyusb#2157] [General]
I know this question might be trivial

Is there a function that can be used to safely remove hardware and eject media for host mode? 

This might be very useful when dealing with MSC class

Maybe there is but I am not able to spot it.

Thank you

**Comment by hathach** [comment]
The procedure is sending SCSI start stop command with correct argument (load_eject = 1, stop = 1), then don't schedule any transfer from usbh, and/or abort any transfer that is queueing, though xfer abort API is not available yet (will add later).

Note: you may also want to TEST UNIT READY first before start stop to make sure device is not writing etc.. That is what is in my head now, you may want to check linux code, observe windows driver and/or checking scsi specs for detail.

**Comment by abakosh** [comment]
Thank you, I will try to implement this and see .
